File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        for(int i=0, m, j=nums.size()-1; i<=j; ) {
            m = i + (j-i)/2;
            if(nums[m] == target) return m;
            if(nums[m] > target) j = m-1;
            else i = m+1;
        }
        return -1;
    }
};

File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        s, e, m = 0, len(nums)-1, 0
        
        while s <= e:
            m = (s + e) // 2
            if nums[m] == target: return m
            elif nums[m] > target: e = m - 1
            else: s = m + 1
        
        return -1 if s > e else m